From c35cd5966eff3a4dc2abd0148d0c2ea3b2fbb6d7 Mon Sep 17 00:00:00 2001
From: Matt Sarnoff <matt@glowforge.com>
Date: Thu, 21 May 2020 23:35:21 -0700
Subject: [PATCH] mxc: ipu3: don't logspam when CONFIG_PM is off

don't dump stack if pm_runtime_get/put fail, which can happen if power
management is disabled.
---
 drivers/mxc/ipu3/ipu_common.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/mxc/ipu3/ipu_common.c b/drivers/mxc/ipu3/ipu_common.c
index c5b82f78421f..314cb98fe8d5 100644
--- a/drivers/mxc/ipu3/ipu_common.c
+++ b/drivers/mxc/ipu3/ipu_common.c
@@ -741,6 +741,7 @@ int32_t ipu_init_channel(struct ipu_soc *ipu, ipu_channel_t channel, ipu_channel
 
 	dev_dbg(ipu->dev, "init channel = %d\n", IPU_CHAN_ID(channel));
 
+#ifdef CONFIG_PM
 	ret = pm_runtime_get_sync(ipu->dev);
 	if (ret < 0) {
 		dev_err(ipu->dev, "ch = %d, pm_runtime_get failed:%d!\n",
@@ -753,6 +754,7 @@ int32_t ipu_init_channel(struct ipu_soc *ipu, ipu_channel_t channel, ipu_channel
 	 * already 'active', so clear it to be 0.
 	 */
 	ret = 0;
+#endif
 
 	_ipu_get(ipu);
 
@@ -1092,7 +1094,7 @@ void ipu_uninit_channel(struct ipu_soc *ipu, ipu_channel_t channel)
 	uint32_t in_dma, out_dma = 0;
 	uint32_t ipu_conf;
 	uint32_t dc_chan = 0;
-	int ret;
+	__maybe_unused int ret;
 
 	mutex_lock(&ipu->mutex_lock);
 
@@ -1306,12 +1308,14 @@ void ipu_uninit_channel(struct ipu_soc *ipu, ipu_channel_t channel)
 
 	_ipu_put(ipu);
 
+#ifdef CONFIG_PM
 	ret = pm_runtime_put_sync_suspend(ipu->dev);
 	if (ret < 0) {
 		dev_err(ipu->dev, "ch = %d, pm_runtime_put failed:%d!\n",
 				IPU_CHAN_ID(channel), ret);
 		dump_stack();
 	}
+#endif
 
 	WARN_ON(ipu->ic_use_count < 0);
 	WARN_ON(ipu->vdi_use_count < 0);
